problems/service-placement/problem-statement-sp.md

```diff
@@ -0,0 +1,146 @@
+
+## Task
+
+Given a graph, a set of requests and their services to be offloaded, minimise the overall latency and processing time of all requests and services, taking resource constraints into account.
+
+## Detailed description
+
+Given **N** nodes, **E** edges, **R** requests and **S$_r$** services for each request $r$, the following objective function aims to minimise latency and processing time:
+
+$\min \sum_{r \in R} \left(\sum_{(u,v) \in E} x_{uvr}l_{uv} + \sum_{i \in N} \sum_{s \in S_r} y_{irs}p_i \right)$
+
+where $x_{er}$ is a 1 if request $r$ traverses edge $e$, 0 otherwise. $y_{irs}$ is 1 if service $s$ from request $r$ is offloaded to node $i$, 0 otherwise. $l_e$ and $p_i$ are the latency of edge $e$ and processing time of node $i$, respectively.
+
+The first constraints aim to ensure that the offloaded services do not exceed the total CPU and RAM. The equations are as follows:
+
+$\sum_{r \in R} \sum_{s \in S_r} y_{irs} cpu_{rs} \leq C_i, ~~~~~\forall i \in N$
+$\sum_{r \in R} \sum_{s \in S_r} y_{irs} ram_{rs} \leq M_i, ~~\forall i \in N$
+
+where $cpu_{rs}$ and $ram_{rs}$ correspond to the requested CPU and RAM, respectively. $C_i$ and $M_i$ correspond to the total CPU and RAM of each node.
+
+It is also necessary to ensure that the total bandwidth of each link is not exceeded. The constraint is as follows:
+
+$\sum_{r \in R} x_{uvr} b_{uv} \leq BW_i, ~~\forall (u,v) \in E$
+
+where $b_e$ corresponds to the requested bandwidth and $BW_i$ corresponds to the total bandwidth of an edge.
+
+Finally, it is necessary to ensure that a path exists. The equation is as follows:
+
+$\sum_{j \in N} x_{ijr} - \sum_{j \in N} x_{jir} = U_{ir}, ~~\forall i \in N, r \in R$
+
+where $U_{ir}$ is 1 if node $i$ corresponds to a source node, -1 if $i$ corresponds to a target node and 0 otherwise.
+
+## Instance data file
+
+Each instance file contains a graph, the requests, and their services requirements.
+
+The first line contains **N** number of nodes.
+Each node contains its number, the total CPU (in MIPS), the total RAM (in Mb), the energy consumption of the processing (in Watt), and the processing time (in s).
+
+After the nodes, the first line contains **E** number of edges.
+Each edge contains the connected nodes, the bandwidth (in Mbps), the energy consumption of the communication (in Watt), and the latency (in s).
+
+Then, the requests start.
+The first line contains **R** number of requests.
+Each request contains the source node, the destination node, and the requested bandwidth.
+
+Afterwards, the services for each request are described.
+The first line contains **S$_r$** number of services for each given request.
+Each service contains the requested CPU (in MIPS) and RAM (in Mb).
+
+Constraints:
+**N** $\geq 2$
+**E** $\geq$**N**$-1$

```

Comment:
Does this constraint mean that the graph must be connected? Is this a restriction of the problem? From my interpretation it would make sense that the nodes in the requests should be connected (otherwise they can never be serviced). However, this does not mean that the whole graph must be connected, right?

In any case if there is some expectation on graph connection it would make sense to write it explicitly somewhere.